otx/cli/tools/train.py

```diff
@@ -144,15 +135,50 @@ def parse_args():
         help="Total number of workers in a worker group.",
     )
 
-    add_hyper_parameters_sub_parser(parser, hyper_parameters)
-
-    return parser.parse_args(), template, hyper_parameters
+    return parser
 
 
-def main():  # pylint: disable=too-many-branches
+# pylint: disable=too-many-branches, too-many-statements
+def main():
     """Main function that is used for model training."""
-    # Dynamically create an argument parser based on override parameters.
-    args, template, hyper_parameters = parse_args()
+    parser = get_parser()
+
+    args, _ = parser.parse_known_args()
+    train_workspace_path = os.getcwd()
+
+    has_template_yaml = Path(args.template).exists()
+    has_data_yaml = Path(args.data).exists()
+    was_built_before = has_template_yaml and has_data_yaml
+    if not was_built_before:
+        # Prepare build
+        train_workspace_path = os.path.join(args.save_model_to)
+        template = find_and_parse_model_template(args.template) if has_template_yaml else None
+
+        # Build
+        builder = Builder()
+        build(
+            builder=builder,
+            train_data_roots=args.train_data_roots,
+            val_data_roots=args.val_data_roots,
+            workspace_root=train_workspace_path,
+            otx_root=get_otx_root_path(),
+            template=template,
+        )
+
+    # Update configurations made by builder
+    # When users input template argument, need to overwrite
+    if args.template != "./template.yaml":
```

Comment:
Could lines 152-165 have anything to do with this code?
If so, I think it would be okay to attach this to line 167 as an else.

Reply:
We should separate this line as is.

Here are the possible use cases.

1. `otx build` --> `otx train --train-data-roots={COCO}`
2. `otx build` --> `otx train --train-data-roots={COCO} --template={SSD}`
3. `otx train --train-data-roots={COCO}`

So, the first thing we have to do is to check whether the workspace exists or not. (`if not was_built_before`).

Let's assume the scenario2, then we need to check the user's input argument (`args.template`).
In this scenario, `otx build` made a `template.yaml` that used ATSS model.
If user gives SSD template to OTX (i.e. `otx train --template={SSD}`), we need to train the model with SSD template.

To be honset, imagining all use cases is quite complex, so if you have any other idea, please let me know

Reply:
What you said seems to be a problem with the current structure. The use case definition we're trying to provide doesn't seem right. For this, we need to proceed with the refactoring we discussed. :)

Reply:
Yes, we should define the discrete use cases to design the system. 
But, I think the above 3 use cases can be included as a real use case.

Reply:
Yes, but I don't think we need to satisfy all three of them in this PR. :)
First of all, this PR should only be considered number 3.

Reply:
Agree with you, although I tried to handle all 3 cases (maybe there are other use cases), I might miss smth.
Let's define the exact use cases and refactoring